embedded-graphics/src/primitives/ellipse.rs

```diff
@@ -0,0 +1,459 @@
+    }
+
+    /// Return the center point of the ellipse scaled by a factor of 2
+    ///
+    /// This method is used to accurately calculate the outside edge of the ellipse.
+    /// The result is not equivalent to `self.center() * 2` because of rounding.
+    fn center_2x(&self) -> Point {
+        let radius = self.size.saturating_sub(Size::new(1, 1));
+
+        self.top_left * 2 + radius
+    }
+}
+
+impl Primitive for Ellipse {
+    type PointsIter = Points;
+
+    fn points(&self) -> Self::PointsIter {
+        Points::new(self)
+    }
+}
+
+impl ContainsPoint for Ellipse {
+    fn contains(&self, point: Point) -> bool {
+        size_to_threshold(
+            Size::new(self.size.width.pow(2), self.size.height.pow(2)),
+            point * 2 - self.center_2x(),
+            self.size.width,
+        )
+    }
+}
+
+impl Dimensions for Ellipse {
+    fn bounding_box(&self) -> Rectangle {
+        Rectangle::new(self.top_left, self.size)
+    }
+}
+
+impl Transform for Ellipse {
+    /// Translate the ellipse from its current position to a new position by (x, y) pixels,
+    /// returning a new `Ellipse`. For a mutating transform, see `translate_mut`.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Ellipse;
+    /// # use embedded_graphics::prelude::*;
+    /// let ellipse = Ellipse::new(Point::new(5, 10), Size::new(10, 15));
+    /// let moved = ellipse.translate(Point::new(10, 10));
+    ///
+    /// assert_eq!(moved.top_left, Point::new(15, 20));
+    /// ```
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            top_left: self.top_left + by,
+            ..*self
+        }
+    }
+
+    /// Translate the ellipse from its current position to a new position by (x, y) pixels.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Ellipse;
+    /// # use embedded_graphics::prelude::*;
+    /// let mut ellipse = Ellipse::new(Point::new(5, 10), Size::new(10, 15));
+    /// ellipse.translate_mut(Point::new(10, 10));
+    ///
+    /// assert_eq!(ellipse.top_left, Point::new(15, 20));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.top_left += by;
+
+        self
+    }
+}
+
+/// Iterator over all points inside the ellipse
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct Points {
+    ellipse: Ellipse,
+    iter: super::rectangle::Points,
+}
+
+impl Points {
+    fn new(ellipse: &Ellipse) -> Self {
+        Self {
+            ellipse: *ellipse,
+            iter: ellipse.bounding_box().points(),
+        }
+    }
+
+    fn empty() -> Self {
+        Self {
+            ellipse: Ellipse::new(Point::zero(), Size::zero()),
+            iter: Rectangle::new(Point::zero(), Size::zero()).points(),
+        }
+    }
+}
+
+impl Iterator for Points {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let ellipse = self.ellipse;
+
+        self.iter.find(|point| ellipse.contains(*point))
```

Comment:
The performance of this could be improved by storing `center_2x` and the squared size in the `Points` struct and using `size_to_threshold`, like in the `StyledEllipseIterator`.